src/components/cuda/cupti_profiler.c

```diff
@@ -0,0 +1,1564 @@
+static int num_gpus;
+static list_metrics_t *avail_events;
+
+static int load_cupti_perf_sym(void);
+static int unload_cupti_perf_sym(void);
+static int load_nvpw_sym(void);
+static int unload_nvpw_sym(void);
+static int initialize_cupti_profiler_api(void);
+static int deinitialize_cupti_profiler_api(void);
+static int initialize_perfworks_api(void);
+static int get_chip_name(int dev_num, char* chipName);
+static int add_events_per_gpu(ntv_event_table_t *event_names, cuptip_control_t *state);
+static int retrieve_metric_details(NVPA_MetricsContext *pMetricsContext, const char *nv_name,
+                                   char *description, int *numDep, NVPA_RawMetricRequest **pRMR);
+static int get_event_names_rmr(NVPA_MetricsContext* pMetricsContext, cuptip_gpu_control_t *gpu_ctl);
+static int check_num_passes(struct NVPA_RawMetricsConfig *pRawMetricsConfig, int rmr_count,
+                            NVPA_RawMetricRequest *rmr, int *num_pass);
+static int control_state_validate(cuptip_control_t *state);
+static int get_counter_availability(cuptip_gpu_control_t *gpu_ctl);
+static int metric_get_config_image(const char *chipName, cuptip_gpu_control_t *gpu_ctl);
+static int metric_get_counter_data_prefix_image(const char* chipName, cuptip_gpu_control_t *gpu_ctl);
+static int create_counter_data_image(cuptip_gpu_control_t *gpu_ctl);
+static int reset_cupti_prof_config_images(cuptip_gpu_control_t *gpu_ctl);
+static int begin_profiling(cuptip_gpu_control_t *gpu_ctl);
+static int end_profiling(cuptip_gpu_control_t *gpu_ctl);
+static int eval_metric_values_per_gpu(NVPA_MetricsContext* pMetricsContext, cuptip_gpu_control_t *gpu_ctl);
+static int get_metric_values(cuptip_control_t *state);
+static int find_same_chipname(int gpu_id);
+static int init_all_metrics(void);
+static void free_all_enumerated_metrics(void);
+static enum collection_method_e get_event_collection_method(const char *evt_name);
+
+NVPA_Status ( *NVPW_GetSupportedChipNamesPtr ) (NVPW_GetSupportedChipNames_Params* params);
+NVPA_Status ( *NVPW_CUDA_MetricsContext_CreatePtr ) (NVPW_CUDA_MetricsContext_Create_Params* params);
+NVPA_Status ( *NVPW_MetricsContext_DestroyPtr ) (NVPW_MetricsContext_Destroy_Params * params);
+NVPA_Status ( *NVPW_MetricsContext_GetMetricNames_BeginPtr ) (NVPW_MetricsContext_GetMetricNames_Begin_Params* params);
+NVPA_Status ( *NVPW_MetricsContext_GetMetricNames_EndPtr ) (NVPW_MetricsContext_GetMetricNames_End_Params* params);
+NVPA_Status ( *NVPW_InitializeHostPtr ) (NVPW_InitializeHost_Params* params);
+NVPA_Status ( *NVPW_MetricsContext_GetMetricProperties_BeginPtr ) (NVPW_MetricsContext_GetMetricProperties_Begin_Params* p);
+NVPA_Status ( *NVPW_MetricsContext_GetMetricProperties_EndPtr ) (NVPW_MetricsContext_GetMetricProperties_End_Params* p);
+NVPA_Status ( *NVPW_CUDA_RawMetricsConfig_CreatePtr ) (NVPW_CUDA_RawMetricsConfig_Create_Params*);
+
+NVPA_Status ( *NVPW_RawMetricsConfig_DestroyPtr ) (NVPW_RawMetricsConfig_Destroy_Params* params);
```

Comment:
Be consistent with spaces

Reply:
Function pointer names are within spaces, the signatures are not. How would you like it?

Reply:
But when you call the function you use spaces. This seems inconsistent.